src/core/__tests__/config.test.tsx

```diff
@@ -0,0 +1,72 @@
+import config, { getOption } from "../../core/config";
+import { mount, React} from "./setup";
+import { TestComponent, WidgetClass } from "./test-component";
+
+class ComponentWithTemplates extends TestComponent {
+    protected _templateProps = [{
+        tmplOption: "item",
+        render: "itemRender",
+        component: "itemComponent",
+        keyFn: "itemKeyFn"
+    }];
+
+    constructor(props: any) {
+        super(props);
+    }
+}
+
+describe("useLegacyTemplatEngine", () => {
+    const originalValue = getOption("useLegacyTemplatEngine");
+
+    beforeEach(() => {
+        config({ useLegacyTemplatEngine: true });
+    });
+
+    afterEach(() => {
+        config({ useLegacyTemplatEngine: originalValue });
+    });
+
+    it("works for render template", () => {
+        const ItemTemplate = (data: any) => (
+            <div className={"template"}>
+                value: {data.value}, key: {data.key}, dxkey: {data.dxkey}
+            </div>
+        );
+
+        const component = mount(
+            <ComponentWithTemplates itemRender={ItemTemplate} />
+        );
+
+        const render = WidgetClass.mock.calls[0][1].integrationOptions.templates.item.render;
```

Comment:
What about reusing the `renderTemplate` function from template tests? Maybe extract is to a test-helpers module

Reply:
Not sure. These tests should be removed in the future with the `useLegacyTemplatEngine` option.